joins.sql

```diff
@@ -0,0 +1,105 @@
+FROM users
+    INNER JOIN posts
+    ON users.id = posts.users_id
+WHERE posts.created_at >= '2015-01-01 00:00:00';
+
+-- 6. Create a query to get the post title, post content, and user's username where the user who created the post joined before January 1, 2015
+SELECT posts.title, posts.content, users.username
+FROM users
+    INNER JOIN posts
+    ON users.id = posts.users_id
+WHERE users.created_at < '2015-01-01 00:00:00';
+
+-- 7. Create a query to get the all rows in the comments table, showing post title (aliased as 'Post Title'), and the all the comment's fields
+SELECT comments.id, comments.body, comments.created_at, comments.updated_at, posts.title AS "Post Title"
+FROM comments
+    LEFT JOIN posts
+    ON comments.posts_id = posts.id;
+
+-- 8. Create a query to get the all rows in the comments table, showing post title (aliased as post_title), post url (ailased as post_url), and the comment body (aliased as comment_body) where the post was created before January 1, 2015
+SELECT posts.title AS post_title, posts.url AS post_url, comments.body AS comment_body
+FROM comments
+    LEFT JOIN posts
+    ON comments.posts_id = posts.id
+WHERE posts.created_at < '2015-01-01 00:00:00';
+
+-- 9. Create a query to get the all rows in the comments table, showing post title (aliased as post_title), post url (ailased as post_url), and the comment body (aliased as comment_body) where the post was created after January 1, 2015
+SELECT posts.title AS post_title, posts.url AS post_url, comments.body AS comment_body
+FROM comments
+    LEFT JOIN posts
+    ON comments.posts_id = posts.id
+WHERE posts.created_at > '2015-01-01 00:00:00';
+
+--10. Create a query to get the all rows in the comments table, showing post title (aliased as post_title), post url (ailased as post_url), and the comment body (aliased as comment_body) where the comment body contains the word 'USB'
+SELECT posts.title AS post_title, posts.url AS post_url, comments.body AS comment_body
+FROM comments
+    LEFT JOIN posts
+    ON comments.posts_id = posts.id
+WHERE comments.body LIKE '%USB%';
+
+--11. Create a query to get the post title (aliased as post_title), first name of the author of the post, last name of the author of the post, and comment body (aliased to comment_body), where the comment body contains the word 'matrix' ( should have 855 results )
+SELECT posts.title AS post_title, users.first_name, users.last_name, comments.body AS comment_body
+FROM comments
+    INNER JOIN posts
+    ON comments.posts_id = posts.id
+    INNER JOIN users
+    ON posts.users_id = users.id
+WHERE comments.body LIKE '%matrix%';
+
+--12. Create a query to get the first name of the author of the comment, last name of the author of the comment, and comment body (aliased to comment_body), where the comment body contains the word 'SSL' and the post content contains the word 'dolorum' ( should have 102 results )
+SELECT users.first_name, users.last_name, comments.body AS comment_body
```

Comment:
This returns the first name and last name of the post author not the comment author.